libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp

```diff
@@ -262,6 +262,20 @@ void NavEKF3_core::SelectMagFusion()
     // used for load levelling
     magFusePerformed = false;
 
+    // Handle case where we are using an external yaw sensor instead of a magnetomer
+    if ((frontend->_magCal == 5) && !use_compass()) {
```

Comment:
It will result in the mag being used instead. I'll modify it so that use_compass returns false if (frontend->_magCal == 5) and remove the  && !use_compass() condition from this line.

Reply:
done